web/src/dropdown_widget.ts

```diff
@@ -712,7 +712,12 @@ export class DropdownWidget {
     // dropdowns don't need it. Maybe we can follow a reverse approach in the future.
     render(value?: number | string): void {
         // Check if the value is valid otherwise just render previous value.
-        if (value !== undefined && typeof value === typeof this.current_value) {
+        const current_value_type = this.text_if_current_value_not_in_options
```

Comment:
I'm not sure if this is a bug or if it's explicitly kept this way to discourage calling render with a string.
I made the mistake of passing a string directly to render() in another PR. We want to uniformly use Option objects to keep them translatable.

And yes, currently `setup()` doesn't call `render()` when the current value is `undefined`. Sahil Batra recommended setting `default_id` to "" for that. Would that be sufficient here?